types/validator_set.go

```diff
@@ -63,27 +64,46 @@ func (vals *ValidatorSet) CopyIncrementAccum(times int) *ValidatorSet {
 // IncrementAccum increments accum of each validator and updates the
 // proposer. Panics if validator set is empty.
 func (vals *ValidatorSet) IncrementAccum(times int) {
+	// call IncrementAccum(1) times times:
+	for i := 0; i < times; i++ {
```

Comment:
What are the intended semantics here? Is this used to catch up if the node has been offline and missed rounds or if the node is syncing past blocks? It looks like this will elect proposers (by setting `vals.Proposer`) and then overwrite them, which would make sense for that use case I suppose - a bit more obvious might be `incrementAccum()` returning a new proposer instead of setting it.

Reply:
The only places where we use `IncrementAccum` with `times > 1` indeed seem to be places where we are catching up. See here:
https://github.com/tendermint/tendermint/blob/72f86b5192a513d6eefd21d80f446f2771394717/consensus/state.go#L754
and recently added here:
https://github.com/tendermint/tendermint/blob/72f86b5192a513d6eefd21d80f446f2771394717/state/store.go#L196

Reply:
OK, that makes sense (to me at least) - I still think it would be clearer if `incrementAccum` returned the new proposer, instead of setting it and then overwriting it when called multiple times in a row - but not a huge deal.